environment/client_config.go

```diff
@@ -59,47 +58,29 @@ func (c *ClientConfig) GetRESTConfig() (*rest.Config, error) {
 		return nil, fmt.Errorf("provided QPS value %f must be >0 and <3.4+e38", c.QPS)
 	}
 
-	// If we have an explicit indication of where the kubernetes config lives, read that.
-	if c.Kubeconfig != "" {
-		return c.configFromPath(c.Kubeconfig)
-	}
-
-	// If not, try the in-cluster config.
-	if rc, err := rest.InClusterConfig(); err == nil {
-		return c.applyOverrides(rc), nil
-	}
-
-	// If no in-cluster config, try the default location in the user's home directory.
-	if c, err := c.configFromPath(clientcmd.RecommendedHomeFile); err == nil {
-		return c, nil
-	}
-
-	return nil, errors.New("could not create a valid kubeconfig")
-}
-
-func (c *ClientConfig) configFromPath(path string) (*rest.Config, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
 	overrides := &clientcmd.ConfigOverrides{}
 
+	if c.Kubeconfig != "" {
+		loadingRules.ExplicitPath = c.Kubeconfig
+	}
 	if c.Cluster != "" {
 		overrides.Context = clientcmdapi.Context{Cluster: c.Cluster}
 	} else if c.ServerURL != "" {
 		overrides.ClusterInfo = clientcmdapi.Cluster{Server: c.ServerURL}
 	}
 
 	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
unsure if this error `err` has context about what it was doing - ie. could not create valid kubeconfig.

If it doesn't can we wrap it prior to returning

Reply:
Wrong lines?

Reply:
yeah maybe this is applicable to the block

```go
if err != nil {
  	return nil, err
}
```

I guess my point is - if this function returns low level error messages (ie. yaml syntax error) should we wrap it

```go
return nil, fmt.Errorf("failed to create REST config: %w", err)
```